srcts/src/components/darkModeSwitch.ts

```diff
@@ -0,0 +1,323 @@
+import type { CSSResultGroup } from "lit";
+import { LitElement } from "lit";
+import { css, html } from "lit";
+import { property } from "lit/decorators.js";
+
+// Inspired by:
+// https://web.dev/building-a-theme-switch-component/
+// https://web.dev/patterns/theming/theme-switch/
+// https://github.com/argyleink/gui-challenges/tree/main/theme-switch
+
+export type DarkModeMessageToggle = {
+  method: "toggle";
+  value: DarkModeSwitch["mode"] | undefined;
+};
+
+export class DarkModeSwitch extends LitElement {
+  static isShinyInput = true;
+  static tagName = "bslib-input-dark-mode";
+  static shinyCustomMessageHandlers = {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    "bslib.toggle-dark-mode": ({
+      method,
+      value,
+    }: DarkModeMessageToggle): void => {
+      // Similar to DarkModeSwitch.receiveMessage(), but we directly update the
+      // Bootstrap attribute on the <html> element. Currently, all toggle switches
+      // follow this value.
+
+      if (method !== "toggle") return;
+
+      if (typeof value === "undefined" || value === null) {
+        const current = document.documentElement.dataset.bsTheme || "light";
+        value = current === "light" ? "dark" : "light";
+      }
+
+      document.documentElement.dataset.bsTheme = value;
+    },
+  };
+
+  private attribute = "data-shinytheme";
+
+  @property({ type: String, reflect: true }) mode!: "dark" | "light";
+
+  static styles: CSSResultGroup = [
+    // CSS Variables
+    css`
+      :host {
+        /* open-props.style via shinycomponent */
+        --text-1: var(--text-1-light, var(--gray-8, #343a40));
+        --text-2: var(--text-2-light, var(--gray-7, #495057));
+        --size-xxs: var(--size-1, 0.25rem);
+        --ease-in-out-1: cubic-bezier(0.1, 0, 0.9, 1);
+        --ease-in-out-2: cubic-bezier(0.3, 0, 0.7, 1);
+
```

Comment:
The transition didn't look right and I figured out that it was because these values were missing:

```suggestion
        --ease-out-1: cubic-bezier(0, 0, 0.75, 1);
        --ease-out-3: cubic-bezier(0, 0, 0.3, 1);
        --ease-out-4: cubic-bezier(0, 0, 0.1, 1);

```